src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +268,109 @@ private EmailWrapper generateSubmissionConfirmationEmail(
 
     }
 
+    private EmailWrapper generateSessionLinksRecoveryNonExistingEmailForStudent(String recoveryEmailAddress) {
+        String emailBody;
+        String subject = EmailType.SESSION_LINKS_RECOVERY.getSubject();
+        String recoveryUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSIONS_LINK_RECOVERY_PAGE).toAbsoluteString();
+        emailBody = Templates.populateTemplate(
+                EmailTemplates.SESSION_LINKS_RECOVERY_EMAIL_NOT_FOUND,
+                "${userEmail}", SanitizationHelper.sanitizeForHtml(recoveryEmailAddress),
+                "${supportEmail}", Config.SUPPORT_EMAIL,
+                "${teammateHomePageLink}", Config.APP_URL,
+                "${sessionsRecoveryLink}", recoveryUrl);
+        EmailWrapper email = getEmptyEmailAddressedToEmail(recoveryEmailAddress);
+        email.setSubject(subject);
+        email.setContent(emailBody);
+        return email;
+    }
+
+    private EmailWrapper generateSessionLinksRecoveryEmailForStudent(String recoveryEmailAddress) {
+        String emailBody;
+        String subject = EmailType.SESSION_LINKS_RECOVERY.getSubject();
+
+        Instant endTime = Instant.now();
+        Instant startTime = endTime.minus(Duration.ofDays(180));
+        Map<String, StringBuilder> linkFragmentsMap = new HashMap<>();
+        String studentName = null;
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllFeedbackSessionsWithinTimeRange(startTime, endTime);
+
+        for (FeedbackSessionAttributes session : sessions) {
+            String courseId = session.getCourseId();
+            CourseAttributes course = coursesLogic.getCourse(courseId);
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, recoveryEmailAddress);
+            StringBuilder linksFragmentValue;
+            if (linkFragmentsMap.containsKey(courseId)) {
+                linksFragmentValue = linkFragmentsMap.get(courseId);
+            } else {
+                linksFragmentValue = new StringBuilder(5000);
+            }
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "";
+                String reportUrlHtml = "";
+
+                if (session.isOpened() || session.isClosed()) {
+                    String submitUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSION_SUBMISSION_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    submitUrlHtml = "[<a href=\"" + submitUrl + "\">submission link</a>]";
+                }
+
+                if (session.isPublished()) {
+                    String reportUrl = Config.getFrontEndAppUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_RESULTS_PAGE)
```

Comment:
this link is wrong. From what i see, you are getting `http://localhost:4200/page/instructorFeedbackResultsPage?courseid=idOfTypicalCourse1&fsname=Second+feedback+session`, which doesnt look right. The right URI looks something like this `http://localhost:4200/web/student/sessions/result?courseid=pigshroommushpig.gma-demo0&fsname=Second+team+feedback+session&key=28FFF3471BA5953D60E1A4508D662B237D1328EC3E471724A860C47F67EA07AAEE04E78C9EAB8B587283255F0B3CD8569760F4A6C2E83C7A3B0D64084929B7E9&studentemail=pigshroommushpig%40gmail.com`